src/consensus.rs

```diff
@@ -199,20 +216,48 @@ impl<T: Proposition> Consensus<T> {
         most_votes > self.elders.threshold()
     }
 
-    fn is_super_majority_over_super_majorities(&self, votes: &BTreeSet<SignedVote<T>>) -> bool {
-        let count_of_agreeing_super_majorities = self
-            .count_votes(&BTreeSet::from_iter(
-                votes
-                    .iter()
-                    .filter(|v| v.vote.is_super_majority_ballot())
-                    .cloned(),
-            ))
+    fn get_super_majority_over_super_majorities(
+        &self,
+        votes: &BTreeSet<SignedVote<T>>,
+    ) -> Result<Option<BTreeMap<T, Signature>>> {
+        let sm_votes = BTreeSet::from_iter(
+            votes
+                .iter()
+                .filter(|v| v.vote.is_super_majority_ballot())
+                .cloned(),
+        );
+        if let Some((props, count_of_sm)) = self
+            .count_votes(&sm_votes)
             .into_iter()
-            .map(|(_, count)| count)
-            .max()
-            .unwrap_or(0);
+            .max_by_key(|(_, c)| *c)
```

Comment:
Just wondering, is it possible to have multiple max here? What happens in that case? (eg: 3 bad nodes that vote twice in a split network, 5 vote for A, 5 for B)

Reply:
You could have multiple max's when we are building up to super-majority. i.e. 1 vote for A, 1 vote for B.

But once we pass the super-majority threshold, only one max will be possible (as long as our assumption of < 1/3 faulty nodes holds.)